### PR TITLE
add form docs to client

### PR DESCRIPTION
**Pull request by khanlifts**
### Contributor checklist

- [x] My contribution is **ready to be merged** as is
- [x] I verified the modifications **render properly**
- [x] I used **spell checking**
- [x] I **used the styling and structure aids available** in [Sphinx/ResT](http://www.sphinx-doc.org/en/stable/rest.html). (Links use `` `…`_``, enumerations `#.`, lists `*`, code ``` ``…`` ```/`.. code::`, warnings .. `warning::`, etc.)
- [x] I reread the text keeping in mind that the text has to be **comprehended** fully **by the target audience**

### Description
add description about the parts the client needs to render forms.

### CODE REVIEW

File: framework/client/form.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  redux-forms requires to set the component prop to specify which component should be used to render the form field. To set
+  the ``<Field/>`` prop component the ``<ReduxFormFieldAdapter/>`` is used (point 7).
+
+7. ReduxFormFieldAdapter
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  The ``<ReduxFormFieldAdapter/>`` is the component that gets passed to ``<Field/>`` returns a call to formFieldFactory (point 2). redux-
+  forms allows to pass a factory as component prop. In addition to the props specific to the tocco-client other props from
+  redux-forms are passed as well. These props contain information about the state of the form e.g. if the form was edited or if
+  the user already clicked on a field.
```
Comment (MethodenMann) [inline framework/client/form.rst]: link redux form documentation?

File: framework/client/form.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+formData
+---------
+
+8. formData
+~~~~~~~~~~~~~
+
+  The formData parent directory contains the formData modules. Each module has its actions, reducers, and sagas to handle
```
Comment (MethodenMann) [inline framework/client/form.rst]: Each module *can have* its actions,

File: framework/client/form.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  the store data relating to data from the form.
+  formData then adds the data and methods to the store.
+
+  props:
+
+  - relationEntities
```
Comment (MethodenMann) [inline framework/client/form.rst]: a description for each "prop" would be nice.

Comment (khanlifts) [reply framework/client/form.rst]: IMO the names are self-explanatory. I started with the descriptions and basically repeated what was already obvious by the names.

File: framework/client/form.rst [hunk 4]
```diff
@@ -0,0 +1,155 @@
+Form
+=====
+
```
Comment (MethodenMann) [inline framework/client/form.rst]: Introduction missing. what is this chapter about.

File: framework/client/form.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  .. code:: js
+
+    ‘longitude’: editableValueFactory(‘coordinate’),
+    ‘latitude’: editableValueFactory(‘coordinate’)
+
+
+4. FormFieldFactory
+~~~~~~~~~~~~~~~~~~~~
+
+  FormFieldFactory receives the data for identifying the correct field from the redux-store through a redux container and
```
Comment (MethodenMann) [inline framework/client/form.rst]: A bit confusing. let us discuss this paragraph

File: framework/client/form.rst [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  provides it to ``<ValueField/>``. The react-redux container is FormDataContainer (point 9). ``<ValueField/>`` returns the specific field
+  accessed through the mapping of the given type of the field.
+
+5. typeEditables
+~~~~~~~~~~~~~~~~~
+
+  The typeEditables directory contains files for different field types like ‘remote’. The files contain methods to overwrite
+  the default behaviour. This can be done through methods that get the events, value and options or data for
```
Comment (MethodenMann) [inline framework/client/form.rst]: The files contain methods to overwrite
  the default behaviour `for each type`.

File: framework/client/form.rst [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+  editableValueFactory. The data then is provided to each ``<EditableValue/>`` field.
+
+form
+-----
+
+6. FormBuilder
+~~~~~~~~~~~~~~~
+
+  The ``<FormBuilder/>`` component traverses through the formDefinition data to resolve how to render each element. It returns the
+  form with ``<LayoutComponent/>`` elements, actions and ``<Field/>`` components provided by the third party library `redux-forms <https://redux-form.com/>`_.
```
Comment (MethodenMann) [inline framework/client/form.rst]: Actions as well.

File: framework/client/form.rst [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  props for ``<FormBuilder/>`` are:
+
+  - model
+  - formName
+  - formDefinition
+  - formFieldMapping
+
+
+  2. The prop formFieldMapping is set with defaultMapping (point 2). In defaultMappings exported object the call to
+  editableValueFactory with type ‘remote’ is mapped to the string 'remote'.
```
Comment (MethodenMann) [inline framework/client/form.rst]: `single-remote-field` not remote

File: framework/client/form.rst [hunk 9]
```diff
@@ -0,0 +1,161 @@
+Form
+=====
+
+Introduction
+-------------
+
+  Forms in the tocco-client are an essential part. This documentation describes what main parts play together to make
```
Comment (MethodenMann) [inline framework/client/form.rst]: ...such as de detail or search form

File: framework/client/form.rst [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+  a form work. The three modules are formField, form and formData. Each of them are described by subfiles and components.
+
+formField
```
Comment (MethodenMann) [inline framework/client/form.rst]: i would add a introduction for each "module". Maybe add a screenshot

**Pull request merged**